### DISCUSSION TITLE
to_zarr(path, overwrite=True)?

### DISCUSSION THREAD

**Discussion by raybellwaves** [pydata/xarray#5103] [Ideas] [answered]
Curious if there is a way to [write a zarr store](http://xarray.pydata.org/en/stable/generated/xarray.Dataset.to_zarr.html) and overwrite it if it already exists?

I'm familiar with overwrite=True when [writing parquet files with dask](https://docs.dask.org/en/latest/dataframe-api.html#dask.dataframe.to_parquet)

**Comment by jakirkham** [comment]
Zarr stores use the [`MutableMapping` API]( https://docs.python.org/3/library/collections.abc.html ), which provides a `clear` method. This can be called to delete all key-value pairs in the store

**Comment by raybellwaves** [comment]
In dask it looks as though it simply does `fs.rm(path, recursive=True)` https://github.com/dask/dask/blob/main/dask/dataframe/io/parquet/core.py#L531

I imaging @jakirkham's solution is probably more effective but I can try and put in a PR using the dask method as a first pass.

**Comment by shoyer** [comment, marked as answer]
You can achieve this today by explicitly passing `mode='w'` into `to_zarr()`.

Per [the docstring](http://xarray.pydata.org/en/stable/generated/xarray.Dataset.to_zarr.html), the default behavior is `mode='w-'`, which is "fail if exists"
